### PR TITLE
Update CoreLib path for consolidation

### PR DESCRIPTION
**Pull request by ViktorHofer**
Search&Replace update of the CoreLib sources from $(CommonPath)\CoreLib\ to $(CoreLibDir). $(CoreLibDir) detects automatically adapts to the runtime repo and changes paths accordingly.

### CODE REVIEW

File: src/Common/tests/Common.Tests.csproj [hunk 1]
```diff
@@ -139,48 +139,48 @@
...
+    <Compile Include="$(CoreLibDir)Interop\Unix\System.Native\Interop.PathConf.cs">
       <Link>Common\Interop\Unix\Interop.PathConf.cs</Link>
     </Compile>
     <Compile Include="$(CommonPath)\Interop\Unix\Interop.Libraries.cs">
       <Link>Common\Interop\Unix\Interop.Libraries.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\System\PasteArguments.Unix.cs">
-      <Link>Common\CoreLib\System\PasteArguments.Unix.cs</Link>
+    <Compile Include="$(CoreLibDir)System\PasteArguments.Unix.cs">
+      <Link>System.Private.CoreLib\src\System\PasteArguments.Unix.cs</Link>
```
Comment (ViktorHofer) [inline src/Common/tests/Common.Tests.csproj]: @stephentoub @jkotas do we actually want System.Private.CoreLib\src in the Link paths or just System\...?

Comment (ViktorHofer) [reply src/Common/tests/Common.Tests.csproj]: I decided to remove those.

File: src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj [hunk 2]
```diff
@@ -73,19 +73,19 @@
...
-      <Link>Common\CoreLib\Interop\Unix\Interop.IOErrors.cs</Link>
+    <Compile Include="$(CoreLibDir)Interop\Unix\Interop.IOErrors.cs">
+      <Link>Interop\Unix\Interop.IOErrors.cs</Link>
     </Compile>
     <Compile Include="$(CommonPath)\System\IO\PathInternal.Unix.cs">
       <Link>Common\System\IO\PathInternal.Unix.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Unix\System.Native\Interop.PathConf.cs">
+    <Compile Include="$(CoreLibDir)Interop\Unix\System.Native\Interop.PathConf.cs">
       <Link>Common\Interop\Unix\Interop.PathConf.cs</Link>
```
Comment (safern) [inline src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj]: It seems like this Link didn't contain  `CoreLib` so it still contains the `Common` prefix.

File: src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj [hunk 3]
```diff
@@ -73,19 +73,19 @@
...
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Unix\System.Native\Interop.MountPoints.cs">
+    <Compile Include="$(CoreLibDir)Interop\Unix\System.Native\Interop.MountPoints.cs">
       <Link>Common\Interop\Unix\Interop.MountPoints.cs</Link>
```
Comment (safern) [inline src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj]: Ditto.

File: src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj [hunk 4]
```diff
@@ -44,7 +44,7 @@
     <Compile Include="System\IO\FileSystemWatcher.Win32.cs">
       <SubType>Component</SubType>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Windows\Kernel32\Interop.CloseHandle.cs">
+    <Compile Include="$(CoreLibDir)Interop\Windows\Kernel32\Interop.CloseHandle.cs">
       <Link>Common\Interop\Windows\Interop.CloseHandle.cs</Link>
```
Comment (safern) [inline src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj]: Ditto.

File: src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj [hunk 5]
```diff
@@ -58,19 +58,19 @@
     <Compile Include="$(CommonPath)\Interop\Unix\Interop.Libraries.cs">
       <Link>Common\Interop\Unix\Interop.Libraries.cs</Link>
     </Compile>
-    <Compile Include="$(CommonPath)\CoreLib\Interop\Unix\System.Native\Interop.Read.cs">
+    <Compile Include="$(CoreLibDir)Interop\Unix\System.Native\Interop.Read.cs">
       <Link>Common\Interop\Unix\libc\Interop.Read.cs</Link>
```
Comment (safern) [inline src/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj]: Ditto. I guess we don't want to fix these? There are many like this, seem to be out of the scope of this PR?

**Review by safern — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Yes that's out of scope.

**Comment by stephentoub** [comment]
Looks good to me.  I might have chosen a name more like CoreLibSharedDir, but you've merged, and it's not a big deal.

**Pull request merged**